pkg/config/remote/data/file.go

```diff
@@ -0,0 +1,64 @@
+package data
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// matches <string>/<int>/<string>/<string>/<string> for <type>/<org_id>/<product>/<config_id>/<file>
+	filePathRegexp       = regexp.MustCompile(`^([^/]+)/(\d+)/([^/]+)/([^/]+)/([^/]+)$`)
+	filePathRegexpGroups = 5
+)
+
+// Type is the global type the file belongs to
+type Type uint
+
+const (
+	// TypeUnknown is an unknown type
+	TypeUnknown Type = iota
+	// TypeDatadog is the datadog type
+	TypeDatadog
+)
+
+// PathMeta contains the metadata of a specific file contained in its path
+type PathMeta struct {
+	Type     Type
+	OrgID    int64
+	Product  Product
+	ConfigID string
+	Name     string
+}
+
+// ParseFilePathMeta parses a file path meta
+func ParseFilePathMeta(path string) (PathMeta, error) {
+	matchedGroups := filePathRegexp.FindStringSubmatch(path)
+	if len(matchedGroups) != filePathRegexpGroups+1 {
+		return PathMeta{}, fmt.Errorf("config file path '%s' has wrong format", path)
+	}
+	rawType := matchedGroups[1]
+	configType := TypeUnknown
+	switch rawType {
+	case "datadog":
+		configType = TypeDatadog
```

Comment:
if it's unknown shouldn't this return an error?

Reply:
The idea is to ignore anything that is not `datadog/` hence the `unknown`